tests/integration/workspace_access/test_redash.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from datetime import timedelta
+
+from databricks.sdk.errors import NotFound
+from databricks.sdk.retries import retried
+from databricks.sdk.service import sql
+
+from databricks.labs.ucx.workspace_access import redash
+from databricks.labs.ucx.workspace_access.groups import GroupManager
+from databricks.labs.ucx.workspace_access.manager import PermissionManager
+from databricks.labs.ucx.workspace_access.redash import RedashPermissionsSupport
+
+logger = logging.getLogger(__name__)
+
+
+# Redash group permissions are cached for up to 10 mins.
+# therefore we need to allow at least 10 mins for checking the permissions after group rename.
+# On top, we have to allow some time to execute the rest of the test.
+@retried(on=[NotFound], timeout=timedelta(minutes=13))
+def test_redash_replace_workspace_groups_with_account_groups(
+    ws,
+    sql_backend,
+    inventory_schema,
+    make_ucx_group,
+    make_user,
+    make_query,
+    make_query_policy_permissions,
+):
+    """
+
+    Args:
+        sql_backend (object):
+    """
+    ws_group, _ = make_ucx_group()
+    user = make_user()
+
+    query = make_query()
+    make_query_policy_permissions(
+        object_id=query.id,
+        permission_level=sql.PermissionLevel.CAN_RUN,
+        group_name=ws_group.display_name,
+        user_name=user.display_name,
+    )
+    logger.info(f"Query: {ws.config.host}/sql/editor/{query.id}")
+
+    group_manager = GroupManager(sql_backend, ws, inventory_schema, [ws_group.display_name], "ucx-temp-")
+    redash_permissions = RedashPermissionsSupport(
+        ws,
+        [redash.Listing(ws.queries.list, sql.ObjectTypePlural.QUERIES)],
+    )
+    permission_manager = PermissionManager(sql_backend, inventory_schema, [redash_permissions])
+
+    permission_manager.inventorize_permissions()
```

Comment:
as i said before, don't call `inventorize_permissions` here, it's too slow.

use the `apply_tasks` instead

```
    apply_tasks(
        generic_permissions,
        [
            MigratedGroup.partial_info(group_a, group_b),
        ],
    )
```

Reply:
but this will both crawl the permissions and apply them to the account. We need to do a rename in between.

Reply:
why do we need to do a rename? you need group_a and group_b and verify if permissions copied from group a to group b.

we have other tests that check if group renaming works.

Reply:
the bug happens when we rename the group and then read redash permissions. It does not happen if you don't do rename.

Reply:
updated